### PR TITLE
New 'move' mutation to re-position elements on the page

### PR DESCRIPTION
**Pull request by bttf**

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -44,4 +62,5 @@ interface ElementRecord {
   attributes: {
     [key: string]: AttributeRecord;
   };
+  position?: MoveRecord;
```
Comment (bryce-fitzsimons) [inline src/types.ts]: should this be nullable instead of optional (to reduce package size)?

File: src/index.ts [hunk 2]
```diff
@@ -88,12 +97,39 @@ function classMutationRunner(record: ClassnameRecord) {
...
+    insertBeforeNode,
+  };
+}
+
+function moveMutationRunner(record: MoveRecord) {
+  let val = record.originalValue;
+  record.mutations.forEach(m => {
+    const selectors = m.mutate();
+    const newNodes = _loadDOMNodes(selectors);
+    val = newNodes ?? val;
```
Comment (jdorn) [inline src/index.ts]: I think null coalescing tends to produce a large bundle size.  In this case, `newNodes` is either an object or `null`, so this is equivalent
```ts
val = newNodes || val;
```

File: src/index.ts [hunk 3]
```diff
@@ -408,11 +493,14 @@ export type DeclarativeMutation = {
...
   value?: string;
+  parentSelector?: string;
+  insertBeforeSelector?: string;
 };
 
 export default {
   html,
   classes,
   attribute,
+  move,
```
Comment (jdorn) [inline src/index.ts]: I'm thinking we get rid of all uses of the word `move` and use `position` instead to keep naming consistent.  So `positionMutationRunner`, `getElementPositionRecord`, etc.

File: src/index.ts (hunk 2 above)
Comment (bttf) [reply src/index.ts]: Updated

File: src/index.ts (hunk 3 above)
Comment (bttf) [reply src/index.ts]: Updated

**Pull request merged**